### ISSUE TITLE
Deck deployment with SSL fails under halyard v1.1.0

### ISSUE THREAD

**Issue by laurieodgers** [spinnaker/spinnaker#2733]
### Issue Summary:
Deploying deck via halyard with SSL enabled `hal config security ui ssl enable` in Kubernetes causes the deck pod to fail to start, resulting in a crashLoopBackoff.

### Cloud Provider(s):
GCP (GKE)

### Environment:
Kubernetes 1.8
Spinnaker 1.7.2
Halyard 1.1.0-20180425120405
Deploying from an Ubuntu 14.04 VM under virtualbox

### Feature Area: 
Deck

### Description:
Expected: Apache under Deck container starts with SSL enabled

Actual: Apache under Deck container fails to start due to "no listening sockets available" (see Additional details for log)

I have attempted to override the port by setting `port: 9000` in ${HOME}/.hal/default/service-settings/deck.yml but this makes no difference. The overridden port makes it through as an environment variable yet deck still fails to start. Additionally, setting port: 443 causes Apache to fail to start due to running as a non root user `(13)Permission denied: AH00072: make_sock: could not bind to address 0.0.0.0:443`

Fix: Editing /etc/apache2/ports.conf and changing `<IfModule ssl_module>`  `<IfModule mod_gnutls.c>` `Listen` lines from 443 to our desired port of 9000 appears to allow apache2 to start.

### Steps to Reproduce:
Use the option `hal config security ui ssl enable` when configuring halyard. Setting this to `disable` allows deck to start as normal (but of course ssl is disabled)

### Additional Details:
```
$ kubectl logs spin-deck-v000-dcwlh -n spinnaker
Considering dependency setenvif for ssl:
Module setenvif already enabled
Considering dependency mime for ssl:
Module mime already enabled
Considering dependency socache_shmcb for ssl:
Enabling module socache_shmcb.
Enabling module ssl.
See /usr/share/doc/apache2/README.Debian.gz on how to configure SSL and create self-signed certificates.
To activate the new configuration, you need to run:
  service apache2 restart
Enabling site spinnaker.
To activate the new configuration, you need to run:
  service apache2 reload
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 10.60.2.22. Set the 'ServerName' directive globally to suppress this message
(13)Permission denied: AH00072: make_sock: could not bind to address [::]:443
(13)Permission denied: AH00072: make_sock: could not bind to address 0.0.0.0:443
no listening sockets available, shutting down
AH00015: Unable to open logs
Action '-D FOREGROUND' failed.
The Apache error log may have more information.
```

Logs from `hal deploy apply` - gets stuck on Deploying spin-deck due to crashLoopBackoff
```
+ Get current deployment
  Success
+ Prep deployment
  Success
+ Preparation complete... deploying Spinnaker
+ Get current deployment
  Success
* Apply deployment
  Determining status of spin-rosco: Waiting on deployments to complete
+ Deploy spin-clouddriver
  Success
+ Deploy spin-front50
  Success
+ Deploy spin-orca
  Success
* Deploy spin-deck
  Deploying spin-deck via provider API: Manually deploying spin-deck
+ Deploy spin-echo
  Success
+ Deploy spin-gate
  Success
+ Deploy spin-igor
  Success
+ Deploy spin-rosco
  Success
```

/etc/apache2/sites-available/spinnaker.conf:
```
<VirtualHost 0.0.0.0:9000>
  SSLEngine on
  SSLCertificateFile "/home/(redacted)/.hal/default/staging/dependencies/248642601-server.crt"
  SSLCertificateKeyFile "/home/(redacted)/.hal/default/staging/dependencies/248649891-server.key"

  DocumentRoot /opt/deck/html

  ProxyPass "/gate" "https://(redacted)" retry=0
  ProxyPassReverse "/gate" "https://(redacted)"

  <Directory "/opt/deck/html/">
    Require all granted
  </Directory>
</VirtualHost>
```

Broken ports.conf:
```
Listen 0.0.0.0:9000

<IfModule ssl_module>
        SSLPassPhraseDialog exec:/etc/apache2/passphrase
        Listen 443
</IfModule>

<IfModule mod_gnutls.c>
        Listen 443
</IfModule>
```

Working ports.conf
```
Listen 0.0.0.0:9000

<IfModule ssl_module>
        SSLPassPhraseDialog exec:/etc/apache2/passphrase
        Listen 9000
</IfModule>

<IfModule mod_gnutls.c>
        Listen 9000
</IfModule>
```

**Comment by iiro** [comment]
Hi @laurieodgers. There was another ticket about this (https://github.com/spinnaker/spinnaker/issues/2708) - but it was closed already.

I've bumped into the same issue - also with GKE. Thanks for the workaround. Disabling SSL is not an option for me.

**Comment by sbwsg** [comment]
The fix for this issue has now been released as part of Spinnaker 1.7.3.  Please feel free to create new github issues if the problem continues to affect your deployments after updating.
